frontend/src/tabs/nfts/NFTs.js

```diff
@@ -53,9 +54,39 @@ const NFTsMainSection = (props) => {
 
 const NFTsExpandedSection = (props) => {
   const imageURL = backendUrl + '/nft-images/';
-  // TODO: Implement filters
   const filters = ['hot', 'new', 'top'];
-  const [activeFilter, setActiveFilter] = React.useState(filters[0]);
+  const [activeFilter, setActiveFilter] = useState(filters[0]);
+
+  useEffect(() => {
+    async function fetchNfts() {
+      try {
+        let result;
+        if (activeFilter === 'top') {
+          result = await getTopNftsFn({
+            page: props.allNftPagination.page,
+            pageLength: props.allNftPagination.pageLength
+          });
+        } else {
+          result = await getNftsFn({
+            page: props.allNftPagination.page,
+            pageLength: props.allNftPagination.pageLength
+          });
+        }
+
+        if (result.data) {
+          if (props.allNftPagination.page === 1) {
+            props.setAllNFTs(result.data);
+          } else {
+            props.setAllNFTs([...props.allNfts, ...result.data]);
```

Comment:
For 'top' and other filters, it will be possible to get some of the same NFTs back on new pages due to changes in like counts and things.
Could you have this function only append the NFTs which are not already included ( the token id should be the unique identifier )